src/lib.rs

```diff
@@ -158,18 +176,47 @@ pub trait Miniconf {
         self.string_set(path.split('/').peekable(), data)
     }
 
+    /// Retrieve a serialized settings value from a string path.
+    ///
+    /// # Args
+    /// * `path` - The path to retrieve.
+    /// * `data` - The location to serialize the data into.
+    ///
+    /// # Returns
+    /// The number of bytes used in the `data` buffer for serialization.
     fn get(&self, path: &str, data: &mut [u8]) -> Result<usize, Error> {
         self.string_get(path.split('/').peekable(), data)
     }
 
+    /// Create an iterator to read all possible settings paths.
+    ///
+    /// # Note
+    /// The state vector can be used to resume iteration from a previous point in time. The data
+    /// should be zero-initialized if starting iteration for the first time.
+    ///
+    /// # Template Arguments
+    /// * `TS` - The maximum number of bytes to encode a settings path into.
+    ///
+    /// # Args
+    /// * `state` - A state vector to record iteration state in.
     fn into_iter<'a, const TS: usize>(
         &'a self,
         state: &'a mut [usize],
-    ) -> iter::MiniconfIter<'a, Self, TS> {
-        iter::MiniconfIter {
+    ) -> Result<iter::MiniconfIter<'a, Self, TS>, IterError> {
```

Comment:
I like this solution of checking buffer sizes at creation time of the iterator. Now that we guard the buffer sizes, I think it would make sense to panic if later code detects that we ran out of space, at that point it is an internal library error right? We wouldn't want to hide it by returning `None` earlier than expected. 

Also, may want to consider adding an `into_iter_unchecked()` version in case a user wants to perform the check only once?

Reply:
Technically the user can still hit these cases if they manually call `recurse_paths()`, but I'm fine with making it `unreachable!()`, since that's not the intended use case.